packages/components/src/components/_internal/SwipeableWithContext.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { LegacyRef, createContext, forwardRef, useContext, useMemo, useRef } from "react";
+import { Swipeable, SwipeableProps } from "react-native-gesture-handler";
+
+const SwipeableWithContextInner = (props: SwipeableProps, ref: LegacyRef<Swipeable>) => {
+    const internalRef = useRef<Swipeable>(null);
+    if (typeof ref === "string" || typeof ref === "function")
+        throw new Error("Don't use legacy refs when referencing SwipeableWithContext");
+    const refToUse = ref ?? internalRef;
+    const swipeableMethods: SwipeableMethods = useMemo(
+        () => ({
+            close: logErrorIfUndefined(refToUse.current?.close),
+            openLeft: logErrorIfUndefined(refToUse.current?.openLeft),
+            openRight: logErrorIfUndefined(refToUse.current?.openRight),
+            reset: logErrorIfUndefined(refToUse.current?.reset),
+        }),
+        [refToUse, refToUse.current],
```

Comment:
you can remove the dependency to refToUse.current as mutable values won't cause the component to re-render

Reply:
Fixed